test.py

```diff
@@ -416,9 +416,72 @@ class IntegerQueueArray(QueueArray[int]):
         self.assertRaises(Exception, test_qa.enqueue, 10)
         
     def test_queue_double_stack(self):
-        test_ds = StoreLineDoubleStack()
+        test_ds = StoreLineDoubleStack(size=1_000_000)
         self.helper(test_ds)
 
+    def test_linked_list_queue_jz(self):
+
+        class IntegerQueueLinkedList(QueueLinkedList[int]):
+            pass
+
+        queue = IntegerQueueLinkedList()
+        self.assertRaises(Exception, queue.peek)
+        self.assertRaises(Exception, queue.dequeue)
+
+        for i in range(50):
+            queue.enqueue(i)
+            self.assertEqual(queue.peek(), 0)
+            self.assertEqual(queue.size, i + 1)
+
+        for i in range(50):
+            self.assertEqual(queue.dequeue(), i)
+            self.assertEqual(queue.size, 49 - i)
+
+    def test_array_queue_jz(self):
+
+        class IntegerQueueArray(QueueArray[int]):
+            pass
+
+        queue = IntegerQueueArray(size=100)
+        self.assertRaises(Exception, queue.peek)
+        self.assertRaises(Exception, queue.dequeue)
+
+        for i in range(50):
+            queue.enqueue(i)
+            self.assertEqual(queue.peek(), 0)
+            self.assertEqual(queue.curr_size, i + 1)
+
+        for i in range(50):
+            self.assertEqual(queue.dequeue(), i)
+            self.assertEqual(queue.curr_size, 49 - i)
+
+        for i in range(100):
+            queue.enqueue(i)
+
+        self.assertRaises(Exception, queue.enqueue, 100)
+
+    def test_double_stack_queue_jz(self):
+
+        class IntegerQueueDoubleStack(QueueDoubleStack[int]):
+            pass
+
+        queue = IntegerQueueDoubleStack(size=100)
+        self.assertRaises(Exception, queue.peek)
+        self.assertRaises(Exception, queue.dequeue)
+
+        for i in range(50):
+            queue.enqueue(i)
+            self.assertEqual(queue.peek(), 0)
+            self.assertEqual(queue.curr_size, i + 1)
```

Comment:
I don't think you can reference `curr_size` here. For example, Adi might use some other internal variable called `size` or `size_billy_bob_joe` or whatever else.